src/webots/nodes/WbMesh.cpp

```diff
@@ -0,0 +1,415 @@
+        texCoordIndex[indexCount] = textureCount;
+        ++textureCount;
+      }
+      ++indexCount;
+    }
+  }
+
+  writer.indent();
+  writer << "coord Coordinate {\n";
+  writer.increaseIndent();
+  writer.indent();
+  writer << "point [\n";
+  writer.increaseIndent();
+  writer.indent();
+  for (int i = 0; i < vertexCount; ++i) {
+    if (i != 0)
+      writer << ", ";
+    const int j = 3 * i;
+    writer << vertex[j] << " " << vertex[j + 1] << " " << vertex[j + 2];
+  }
+  writer << "\n";
+  writer.decreaseIndent();
+  writer.indent();
+  writer << "]\n";
+  writer.decreaseIndent();
+  writer.indent();
+  writer << "}\n";
+
+  writer.indent();
+  writer << "normal Normal {\n";
+  writer.increaseIndent();
+  writer.indent();
+  writer << "point [\n";
+  writer.increaseIndent();
+  writer.indent();
+  for (int i = 0; i < normalCount; ++i) {
+    if (i != 0)
+      writer << ", ";
+    const int j = 3 * i;
+    writer << normal[j] << " " << normal[j + 1] << " " << normal[j + 2];
+  }
+  writer << "\n";
+  writer.decreaseIndent();
+  writer.indent();
+  writer << "]\n";
+  writer.decreaseIndent();
+  writer.indent();
+  writer << "}\n";
+
+  writer.indent();
+  writer << "texCoord TextureCoordinate {\n";
+  writer.increaseIndent();
+  writer.indent();
+  writer << "point [\n";
+  writer.increaseIndent();
+  writer.indent();
+  for (int i = 0; i < textureCount; ++i) {
+    if (i != 0)
+      writer << ", ";
+    const int j = 2 * i;
+    writer << texture[j] << " " << texture[j + 1];
+  }
+  writer << "\n";
+  writer.decreaseIndent();
+  writer.indent();
+  writer << "]\n";
+  writer.decreaseIndent();
+  writer.indent();
+  writer << "}\n";
+
+  writer.indent();
+  writer << "coordIndex [\n";
+  writer.increaseIndent();
+  writer.indent();
+  for (int i = 0; i < indexCount; ++i) {
+    if (i != 0) {
+      writer << " ";
+      if (i % 3 == 0)
+        writer << "-1 ";
+    }
+    writer << coordIndex[i];
+  }
+  writer << " -1\n";
+  writer.decreaseIndent();
+  writer.indent();
+  writer << "]\n";
+
+  writer.indent();
+  writer << "normalIndex [\n";
+  writer.increaseIndent();
+  writer.indent();
+  for (int i = 0; i < indexCount; ++i) {
+    if (i != 0) {
+      writer << " ";
+      if (i % 3 == 0)
+        writer << "-1 ";
+    }
+    writer << normalIndex[i];
+  }
+  writer << " -1\n";
+  writer.decreaseIndent();
+  writer.indent();
+  writer << "]\n";
+
+  writer.indent();
+  writer << "texCoordIndex [\n";
+  writer.increaseIndent();
+  writer.indent();
+  for (int i = 0; i < indexCount; ++i) {
+    if (i != 0) {
+      writer << " ";
+      if (i % 3 == 0)
+        writer << "-1 ";
+    }
+    writer << texCoordIndex[i];
+  }
+  writer << " -1\n";
+  writer.decreaseIndent();
+  writer.indent();
+  writer << "]\n";
+
+  delete[] coordIndex;
+  delete[] normalIndex;
+  delete[] texCoordIndex;
+  delete[] vertex;
+  delete[] normal;
+  delete[] texture;
+}
+
+void WbMesh::rescale(const WbVector3 &scale) {
+}
```

Comment:
This empty method is really needed?
If yes, then there is no need to put in in the CPP file but could be defined directly in the HPP